ansible/roles/sshkey-host-access/tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+
+- name: ensure the existence of the user directory
+  file:
+    path: "{{ KEY_LOC_PATH }}"
+    owner: "{{ KEY_LOC_OWNER }}"
+    group: "{{ KEY_LOC_GROUP }}"
+    mode: 0700
+    state: directory
+  delegate_to: "{{ target_host }}"
+
+- block:
+    - name: check the existence of a given ssh key by name
+      stat:
+        path: "{{ FULL_PRIVKEY_PATH }}"
+      register: go_key
+
+    - name: generate the keypair
+      command: >
+        /usr/bin/ssh-keygen -N '' -t rsa -b 2048 -f {{ KEY_LOC_PATH }}/{{ ATMOUSERNAME }}{{ KEY_LOC_PREFIX }}_default -C {{ ATMOUSERNAME }}
+      when: go_key.stat.exists == False
+
+    - name: set the permissions on the private key file
+      file:
+        path: "{{ FULL_PRIVKEY_PATH }}"
+        mode: 0600
+        owner: "{{ KEY_LOC_OWNER }}"
+        group: "{{ KEY_LOC_GROUP }}"
+
+    - name: add the key name to .default_ids
+      lineinfile:
+        dest: "{{ KEY_LOC_PATH }}/.default_ids"
+        line: "{{ ATMOUSERNAME }}{{ KEY_LOC_PREFIX }}_default"
+        create: yes
+        state: present
+        owner: "{{ KEY_LOC_OWNER }}"
+        group: "{{ KEY_LOC_GROUP }}"
+
+    - name: finally, transfer the public key locally
+      fetch:
+        src: "{{ FULL_PUBKEY_PATH }}"
+        dest: "{{ LOCAL_PUBKEY_DIR }}"
+        flat: yes
+
+    - name: Ensure correct ownership of public key
+      file:
+        path: "{{ FULL_PUBKEY_PATH }}"
+        owner: "{{ KEY_LOC_OWNER }}"
+        group: "{{ KEY_LOC_GROUP }}"
+  delegate_to: "{{ target_host }}"
+
+- name: distribute the generated pub key to the instance
+  authorized_key:
+    key: "{{ lookup('file', LOCAL_PUBKEY_DIR + ATMOUSERNAME + KEY_LOC_PREFIX + '_default.pub') }}"
```

Comment:
Does the following work: `lookup('file', FULL_PUBKEY_PATH)`. You do not need to handle failure with `lookup` since you're generating that key, and if it fails to generate, then the error will be caught there.